modules/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/OTelTracerModulePlugin.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing;
+
+import org.opensearch.common.SetOnce;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.FeatureFlags;
+import org.opensearch.plugins.Plugin;
+import org.opensearch.plugins.TracerPlugin;
+import org.opensearch.telemetry.Telemetry;
+import org.opensearch.telemetry.metrics.MetricsTelemetry;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import static org.opensearch.common.util.FeatureFlags.TRACER;
+
+/**
+ * Tracer plugin based on Otel
+ */
+public class OTelTracerModulePlugin extends Plugin implements TracerPlugin {
+
+    static final String OTEL_TRACER_NAME = "otel";
+
+    private SetOnce<Telemetry> telemetry = new SetOnce<>();
```

Comment:
The plugins are initialized only once, why do we need this `SetOnce` ? We just need to return the `OtelTelemetryImpl` instance in supplier.

Reply:
We were having supplier of telemetry earlier to ensure Otel classes are not loaded unless used(in instrumentation). Hence we needed it. But i now think thats not needed as we will have instrumentation always. Removed it in the latest revision